File: tiq-e/array/9/main.py
from time import perf_counter
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {}
        for i in range(len(nums)):
            if nums[i] in d:
                return [d[nums[i]], i]
            else:
                d[target - nums[i]] = i
        return []


arr = [2, 7, 11, 15]

start = perf_counter()
result = Solution().twoSum(arr, 9)
end = perf_counter()
print("Duration: " + str(end - start))
print("Result: " + str(result))

File: tiq-e/array/9/main.go
package main

import (
	"fmt"
	"time"
)

func twoSum(nums []int, target int) []int {

	lookup := make(map[int]int)

	for i, v := range nums {
		j, ok := lookup[v]

		lookup[target-v] = i

		if ok {
			return []int{j, i}
		}

	}
	return []int{}

}

func main() {
	arr := []int{0, 1, 0, 3, 12}

	var start = time.Now()
	twoSum(arr, 9)
	var end = time.Since(start)
	fmt.Println("Duration: ", end)
	fmt.Println("Result: ", arr)
}
